src/coreclr/src/jit/codegenxarch.cpp

```diff
@@ -3005,20 +3011,36 @@ void CodeGen::genCodeForInitBlkUnroll(GenTreeBlk* node)
             // For x86, we need one more to convert it from 8 bytes to 16 bytes.
```

Comment:
This `else` block could be more efficient as just a shuffle (SSE2) or broadcast (AVX2). You can see the pattern we use for setting all elements of a vector to `value` here:
* 32-bit: https://source.dot.net/#System.Private.CoreLib/Vector128.cs,399
* 64-bit: https://source.dot.net/#System.Private.CoreLib/Vector128.cs,434

Likewise for V256 it can be a shuffle + insert (AVX) or broadcast (AVX2):
* 32-bit: https://source.dot.net/#System.Private.CoreLib/Vector256.cs,357
* 64-bit: https://source.dot.net/#System.Private.CoreLib/Vector256.cs,396

Reply:
We start with 2 bytes of data after `emit->emitIns_R_R(INS_mov_i2xmm, EA_PTRSIZE, srcXmmReg, srcIntReg);`, if we want to use a broadcast we will need to copy it up to 4 bytes. It could still be profitable, I think it is worth to open an issue from your comment, mark it as easy and up-for-grabs probably.

Reply:
How do we start with 2 bytes of data? this is `mov_i2xmm` which is `movd` which only supports reading 32-bits or 64-bits (also hence the `EA_PTRSIZE`).

Reply:
oh yes, we are transferring from 1 byte that `Initblk` accepts to int/long/float size in `fgMorphPromoteLocalInitBlock`